chat-plugins/mafia.js

```diff
@@ -0,0 +1,609 @@
+
+		leave: function (target, room, user) {
+			if (!room.game || room.game.gameid !== 'mafia') return this.errorReply("There is no game of mafia running in this room.");
+			if (room.game.gamestate !== 'pregame') return this.errorReply("The game has started already.");
+			if (!this.canTalk()) return this.errorReply("You cannot do this while unable to talk.");
+
+			if (room.game.removePlayer(user)) {
+				room.game.updatePregame();
+			} else {
+				return this.errorReply("You're not in the game.");
+			}
+		},
+
+		display: function (target, room, user) {
+			if (!room.game || room.game.gameid !== 'mafia') return this.errorReply("There is no game of mafia running in this room.");
+			if (room.game.gamestate !== 'pregame') return this.errorReply("The game has started already.");
+			if (!this.canTalk()) return this.errorReply("You cannot do this while unable to talk.");
+
+			room.game.displayPregame();
+		},
+
+		target: function (target, room, user) {
+			if (!room.game || room.game.gameid !== 'mafia') return this.errorReply("There is no game of mafia running in this room.");
+			if (room.game.gamestate === 'pregame') return this.errorReply("The game hasn't started yet.");
+			if (!this.canTalk()) return this.errorReply("You cannot do this while unable to talk.");
+
+			if (user.userid in room.game.players && toId(target) in room.game.players) {
+				room.game.players[user.userid].onReceiveTarget(room.game.players[toId(target)]);
+			}
+		},
+
+		vote: function (target, room, user) {
+			if (!room.game || room.game.gameid !== 'mafia') return this.errorReply("There is no game of mafia running in this room.");
+			if (!room.game.gamestate === 'pregame') return this.errorReply("The game hasn't started yet.");
+			if (!this.canTalk()) return this.errorReply("You cannot do this while unable to talk.");
+
+			if (user.userid in room.game.players && toId(target) in room.game.players) {
+				room.game.players[user.userid].onReceiveVote(room.game.players[toId(target)]);
+			}
+		},
+
+		end: function (target, room, user) {
+			if (!this.can(permission, null, room)) return false;
+			if (!room.game || room.game.gameid !== 'mafia') return this.errorReply("There is no game of mafia running in this room.");
+			if (!this.canTalk()) return this.errorReply("You cannot do this while unable to talk.");
+
+			room.game.forceEnd();
+			return this.privateModCommand("(The game of mafia was forcibly ended by " + user.name + ".)");
+		},
+
+		disable: function (target, room, user) {
+			if (!this.can('tournamentsmanagement', null, room)) return;
+			if (!room.mafiaEnabled) {
+				return this.errorReply("Mafia is already disabled.");
+			}
+			delete room.mafiaEnabled;
+			if (room.chatRoomData) {
+				delete room.chatRoomData.mafiaEnabled;
+				Rooms.global.writeChatRoomData();
+			}
+			return this.sendReply("Mafia has been disabled for this room.");
+		},
+
+		enable: function (target, room, user) {
+			if (!this.can('tournamentsmanagement', null, room)) return;
```

Comment:
Let's give it its own permission; I don't want random people to test it at this point.